docs/configuration/trainer.md

```diff
@@ -148,3 +150,75 @@ centered: false
     clipnorm: null
     clip_value: null
     ```
+
+# Configuring training length
+
+The length of the training process is configured by:
+
+* `epochs` (default: 100): One epoch is one pass through the entire dataset. By
+    default, `epochs` is 100 which means that the training process will run for
+    a maximum of 100 epochs before terminating.
+* `train_steps` (default: `None`): The maximum number of steps to train for,
+    using one mini-batch per step. By default this is unset, and `epochs` will
+    be used to determine training length.
+
+!!! tip
+
+    In general, it's a good idea to set up a long training runway, relying on
+    early stopping criteria (`early_stop`) to stop training when there
+    hasn't been any improvement for a long time.
+
+# Configuring checkpoint and evaluation frequency
+
+Evaluation is run every time the model is checkpointed.
+
+By default, checkpoint-evaluation will occur once every epoch.
+
+The frequency of checkpoint-evaluation can be configured using:
+
+* `steps_per_checkpoint` (default: 0): every `n` training steps
+* `checkpoints_per_epoch` (default: 0): `n` times per epoch
+
+!!! note
+
+    It is invalid to specify both non-zero `steps_per_checkpoint` and non-zero
+    `checkpoints_per_epoch`.
+
+!!! tip
+
+    Running evaluation once per epoch is an appropriate fit for tabular
+    datasets, which are small, fit in memory, and train quickly. However, this 
```

Comment:
Would in general make it more about how often does the user want do evaluate vs the size of the dataset rather than the datatype. If you want to talk about the datatype you can say that it usually influences how fast can you get a single epoch to run and say that usually a single epoch of a tabular dataset is faster than a text or image one

Reply:
Good thought. I've revised the wording to focus on this recommendation for small datasets.